CIPs/0007.md

```diff
@@ -0,0 +1,108 @@
+
+### Variation of the proposed solution
+
+- Every user generates the validation slots for itself. The first user to gather all the validations and trigger the slashing, collects the reward
+
+- The slots validated works for every user in the community. When a slashing is performed, every user that generated a validated slot that is used to demostrate the downtime, receives a portion of the reward 
+
+## Alternative Solutions
+
+### Validator's Uptime in the epoch
+
+The Celo blockchain for reward purposes, calculates an uptime for each validator. The validator’s uptime for the epoch is the proportion of blocks in the epoch for which it is ‘up’: U = counter / (epoch_size - 11).
+
+One possible approach, it's to slash the validators if they are not covering a certain umbral.
+
+#### Pros
+
+The uptime calculation it's part of the Celo protocol. This not only allow us to save gas from calculating a possible slashing, but also could let the same blockchain to perform the slashes. This would make a safer net, instead of requiring a so rigurous control of the users.
+
+In the actual scenario, if no user is controling, no one get slashed.
+
+#### Cons
+
+The validator's uptime is calculated at the end of the epoch, that has two potential problems:
+
+- The slashing for an specific validator will be performed in the next epoch. That allows a validator that is not performing well in the epoch X, to be reelected for the epoch X+1, because the tool used for that (the uptime) was added after the election was completed.
+
+- The uptime is a relation that has meaning tied to an specific epoch as whole. It won't give you a detail information of what happened.
+
+For example, if the community sets the lower umbral as 50% this two scenarios have the same uptime:
+ * A Validator is down for 1/2 epoch consecutive, and then signs every block for the next 1/2 epoch
+ * A Validator signs 1 every 2 packages
+
+Another problem is the calculation between epoch for reelected validators.
+Using the same umbral of the other example, if the validator has in the epoch X = 75% uptime, and in the epoch X+1 = 70% uptime, this two numbers could hide a more that a 1/2 epoch consecutive blocks downtime. This will happen if the 25% of the downtime in the X is at the ending of the epoch, and the other 30% of the downtime for the epoch X+1 is at the beginning. 
+
+### Challenge the validator
+
+This solutions consist in solving a challenge between a user and the validator. 
+
+The user could claim that the validator has been down for an specific period of blocks. In which case, the validator has an specific amount of time to bring a proove where the blockchain could verify that the validator was up. For example 1 or N blocks that where signed by the validator.
+
+#### Pros
+
+From the blockchain perspective, a simple and easy way for validating the uptime.
```

Comment:
Implementing a challenge-response scheme would be far from simple in practice. I don't think we have a fleshed out enough idea of what this would look like to really include it as a proposal in this CIP.